[stable/mongodb-replicaset] liveness/startupProbe tuning

wonko opened the pull request:
#### What this PR does / why we need it:

- Adds a startupProbe to the chart, so a larger mongo install won't be terminated by the livenessProbe (will only apply on semverMatch >= 1.16).
- Updated the metrics container to also use the livenessProbe values instead of hardcoded ones.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

wonko committed:
MongoDB liveness/startupProbe tuning

Signed-off-by: Bernard Grymonpon <bernard.grymonpon@oqton.com>

steven-sheehy commented on stable/mongodb-replicaset/templates/mongodb-statefulset.yaml:
```diff
@@ -221,6 +221,24 @@ spec:
             failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
             periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
             successThreshold: {{ .Values.livenessProbe.successThreshold }}
+{{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion -}}
```
Looks like startupProbe is not enabled by default (e.g. marked beta) until 1.18. Would prefer this be set to that.

steven-sheehy commented on stable/mongodb-replicaset/values.yaml:
```diff
@@ -198,3 +198,12 @@ livenessProbe:
+# Startup probe, will only be added since Kubernetes v1.16 or higher
+# When you have a lot of collections/indices, MongoDB need some time to start
+startupProbe:
+  initialDelaySeconds: 5
```
These don't match the README. Please sync them

steven-sheehy commented on stable/mongodb-replicaset/templates/mongodb-statefulset.yaml:
```diff
@@ -296,8 +314,26 @@ spec:
+          startupProbe:
+            exec:
+              command:
+                - mongo
```
This shouldn't test mongo directly but run the same test as the readiness probe. The readiness probe already tests the mongo connection itself.

steven-sheehy reviewed (changes requested):
/assign
/ok-to-test

wonko committed:
Changes after review

- 1.18 as target k8s api version
- aligned the values and readme
- changed probe for the metrics server

Signed-off-by: Bernard Grymonpon <bernard.grymonpon@oqton.com>

wonko committed:
Merge branch 'master' into feature/add-startupprobe-to-mongo

steven-sheehy reviewed (approved):
/lgtm
